### DISCUSSION TITLE
What is the safest way to get the page number?

### DISCUSSION THREAD

**Discussion by jsbien** [trufanov-nok/djvudict#3] [Q&A]
In my first experiments the directories were named like `100_000584_0090.djvu` and I assumed the last numerical component is the page number. However when preparing the test for my previous question I got the directory named `0_est.djvu`. So what is the logic behind the directory names?

**Comment by trufanov-nok** [comment]
If I recall right:  
It seems to be an `entryNo_entryName_entryTitle` where entry no is a 0-based position in a list of document entries.   
Each document contains a header. Header is a single entry or a list of entries. Multipage doc is always a list. The document structure can be printed out with djvulibre's djvudump tool.   
Entry might be not a page but for ex. a Shared dictionary. Also entry might have Name (usually for multipage) and might have a Title (optional and rarely used).   

In case there is neither name or title (might be for singlepage) - the document filename is used.
I guess djvu doesn't keep page numbers in the header but calculate them on the fly by ignoring everything in entries list that is not a page (DJVU tag) while counting.

![image](https://github.com/trufanov-nok/djvudict/assets/5746456/3be43212-98b0-4d53-84f7-ddf7fcae67a6)

**Comment by trufanov-nok** [comment]
P.S. entry names are usually generated by a djvu encoders and are not customized by user. But minidjvu allows to change djbz "extension" in a name of the entry that contains a shared dictionary  with --Xtension for some reason that noone remember anymore (i guess). But if I recall right DjView4 doesn't like any extensions for shared dictionary entries except .djbz and .iff

**Comment by trufanov-nok** [comment]
I guess the name of the entry (without extension) in a multipage document is set by encoders to either the source image file name or some incremental id by default. They must be unique.

**Comment by jsbien** [comment]
Thank for all the comments.
I consider having a look at the code with a debugger. Which one would you advice to use?
